scripts/run-stylish-haskell.sh

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
+#
+# Run stylish-haskell on a file, using the stylish-haskell.yaml file that is
+# closest to the file in the directory hierarchy.
+
+file="$1"
+
+directory="$(dirname "$file")"
+
+while [[ "$directory" != "/" && "$directory" != "." ]]; do
+  config_file="$directory/.stylish-haskell.yaml"
+  if [[ -f "$config_file" ]]; then
+    stylish_haskell_config_file="$config_file"
+    break
+  fi
+  directory="$(dirname "$directory")"
+done
+
+if [[ -f "$stylish_haskell_config_file" ]]; then
+  stylish-haskell -c "$stylish_haskell_config_file" -i "${file}"
+else
+  stylish-haskell -i "${file}"
+fi
```

Comment:
Usually the repos I used, had `.stylish-haskell.yml` in the repository root. So I guess `(cd $(git root) ; stylish-haskell filename.hs )`  should work there as well I guess. 

Do we have an example somewhere where this script becomes useful?

Reply:
When the repository has multiple `.stylish-haskell.yml` files for example one that uses submodules, `stylish-haskell` doesn't find the closes such file, it just looks at the current directory.